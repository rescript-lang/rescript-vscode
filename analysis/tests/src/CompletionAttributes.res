// @module("") external doStuff: t = "test"
//          ^com

