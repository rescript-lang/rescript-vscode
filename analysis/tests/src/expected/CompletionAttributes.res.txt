Complete src/CompletionAttributes.res 0:8
Attribute id:modu:[0:3->0:8] label:modu
Completable: Cdecorator(modu)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "module",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": {"kind": "markdown", "value": "The `@module` decorator is used to bind to a JavaScript module.\n\n[Read more and see examples in the documentation](https://rescript-lang.org/syntax-lookup#module-decorator)."},
    "insertText": "module(\"$0\")",
    "insertTextFormat": 2
  }]

Complete src/CompletionAttributes.res 3:12
XXX Not found!
Completable: CdecoratorPayload(module=)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "@rescript/react",
    "kind": 4,
    "tags": [],
    "detail": "Package",
    "documentation": null
  }, {
    "label": "./TableclothMap.ml",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./TableclothMap.mli",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./test.json",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./tst.js",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }]

Complete src/CompletionAttributes.res 6:7
Attribute id:js:[6:3->6:7] label:@js
Completable: Cdecorator(@js)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "jsxConfig",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": {"kind": "markdown", "value": "The `@@jsxConfig` decorator is used to change the config for JSX on the fly.\n\n[Read more and see examples in the documentation](https://rescript-lang.org/docs/manual/latest/jsx#file-level-configuration)."},
    "insertText": "jsxConfig({$0})",
    "insertTextFormat": 2
  }]

Complete src/CompletionAttributes.res 9:16
XXX Not found!
Completable: JsxConfig
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "version",
    "kind": 5,
    "tags": [],
    "detail": "int",
    "documentation": {"kind": "markdown", "value": "```rescript\nversion?: int\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }, {
    "label": "module_",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nmodule_?: string\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }, {
    "label": "mode",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nmode?: string\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }]

Complete src/CompletionAttributes.res 12:17
XXX Not found!
Completable: JsxConfig
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "module_",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nmodule_?: string\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }, {
    "label": "mode",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nmode?: string\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }]

Complete src/CompletionAttributes.res 15:25
XXX Not found!
Completable: JsxConfig
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "\"\"",
    "kind": 12,
    "tags": [],
    "detail": "string",
    "documentation": null,
    "sortText": "A",
    "insertText": "\"$0\"",
    "insertTextFormat": 2
  }]

Complete src/CompletionAttributes.res 18:29
XXX Not found!
Completable: JsxConfig
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "version",
    "kind": 5,
    "tags": [],
    "detail": "int",
    "documentation": {"kind": "markdown", "value": "```rescript\nversion?: int\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }, {
    "label": "mode",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nmode?: string\n```\n\n```rescript\ntype jsxConfig = {version: int, module_: string, mode: string}\n```"}
  }]

Complete src/CompletionAttributes.res 21:12
XXX Not found!
Completable: CdecoratorPayload(moduleWithImportAttributes)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "from",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nfrom?: string\n```\n\n```rescript\ntype moduleConfig = {from: string, with: string}\n```"}
  }, {
    "label": "with",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nwith?: string\n```\n\n```rescript\ntype moduleConfig = {from: string, with: string}\n```"}
  }]

Complete src/CompletionAttributes.res 24:17
XXX Not found!
Completable: CdecoratorPayload(moduleWithImportAttributes)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "{}",
    "kind": 12,
    "tags": [],
    "detail": "importAttributesConfig",
    "documentation": {"kind": "markdown", "value": "```rescript\ntype importAttributesConfig = {type_: string}\n```"},
    "sortText": "A",
    "insertText": "{$0}",
    "insertTextFormat": 2
  }]

Complete src/CompletionAttributes.res 27:19
XXX Not found!
Completable: CdecoratorPayload(moduleWithImportAttributes)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "type_",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\ntype_?: string\n```\n\n```rescript\ntype importAttributesConfig = {type_: string}\n```"}
  }]

Complete src/CompletionAttributes.res 30:19
XXX Not found!
Completable: CdecoratorPayload(module=)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "@rescript/react",
    "kind": 4,
    "tags": [],
    "detail": "Package",
    "documentation": null
  }, {
    "label": "./TableclothMap.ml",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./TableclothMap.mli",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./test.json",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./tst.js",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }]

Complete src/CompletionAttributes.res 33:17
XXX Not found!
Completable: CdecoratorPayload(module=)
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
[{
    "label": "@rescript/react",
    "kind": 4,
    "tags": [],
    "detail": "Package",
    "documentation": null
  }, {
    "label": "./TableclothMap.ml",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./TableclothMap.mli",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./test.json",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }, {
    "label": "./tst.js",
    "kind": 4,
    "tags": [],
    "detail": "Local file",
    "documentation": null
  }]

